File: string/462-removeDigit.py
def removeDigit( number, digit):
        """
        :type number: str
        :type digit: str
        :rtype: str
        """
        n=len(number)
        ans=0
        for i in range(n):
            if number[i]==digit:
                ans=max(ans,int(number[:i]+number[i+1:]))
                
        return str(ans)
    
            

File: string/462-removeDigit.cpp
#include<string>
#include<iostream>

using STR=std::string;

STR removeDigit(const STR& s,char digit)
{
    STR ans{"0"};
    int n=s.size();
    for(int i=0;i<n;++i)
    {
        if(s[i]==digit)
        {
            STR temp=s.substr(0,i)+s.substr(i+1,n-i-1);
            ans=std::to_string(std::max(std::stoi(ans),std::stoi(temp)));
        }
    }
    return ans;
}
int main()
{
    STR s{"11"};
    std::cout<<removeDigit(s,'1');
}